frontend/src/metabase/admin/permissions/types.ts

```diff
@@ -37,34 +37,83 @@ export type TableEntityId = SchemaEntityId & {
 export type EntityId = DatabaseEntityId &
   Partial<Omit<TableEntityId, "databaseId">>;
 
-export type DataPermission = "data" | "download" | "data-model" | "details";
+export enum DataPermission {
+  VIEW_DATA = "view-data",
+  CREATE_QUERIES = "create-queries",
+  DOWNLOAD = "download",
+  DATA_MODEL = "data-model",
+  DETAILS = "details",
```

Comment:
For the sake of code prettiness, perhaps ViewData, CreateQueries, etc.?

Reply:
In the JS ecosystem, I think screaming snake case denotes constants and pascal case classes / factory functions. I'm prone to keep it, but I'm also a push over.

Reply:
I wasn't sure of the convention earlier this week and I found this: https://www.typescriptlang.org/docs/handbook/enums.html

Fwiw there is an abandoned RFC about the capitalization of constants https://www.notion.so/metabase/38-Use-camelCase-for-constants-4655f93f9eac4ed59f588b1df4b9d975